src/compiler/parser.ts

```diff
@@ -6698,7 +6699,8 @@ namespace ts {
                             if (typeExpression && typeExpression.type.kind === SyntaxKind.ArrayType) {
                                 jsdocTypeLiteral.isArrayType = true;
                             }
-                            typedefTag.typeExpression = finishNode(jsdocTypeLiteral);
+                            const useChildTypeTagAsType = childTypeTag && !isObjectOrObjectArrayTypeReference(childTypeTag.typeExpression.type);
+                            typedefTag.typeExpression = useChildTypeTagAsType ? childTypeTag.typeExpression : finishNode(jsdocTypeLiteral);

```

Comment:
Probably better to make these one expression so that if we have strictNullChecks TS can infer that we just checked for `childTypeTag` being defined.

Reply:
Good point. Done.